src/solvers/README.md

```diff
@@ -62,7 +62,109 @@ different decision procedures, roughly one per directory.
 
 \section flattening-section Flattening
 
-To be documented.
+**Key classes:**
+* \ref propt
+* \ref boolbvt
+* \ref bv_utils
+
+The main class in this folder is \ref boolbvt which wraps a variety of 
+helper methods and classes (including inherited ones) and acts as an interface 
+for transforming \ref exprt into boolean formula and then solving it.
+
+Many of its methods are focused around transforming a particular \ref exprt 
+into a vector of \ref literalt and then passing them to a \ref propt 
+for formula building / solving.
+
+The primary methods are:
+
+`bvt boolbvt::convert_bitvector(const exprt &expr)`
+
+Which takes an exprt then calls the associated transformation functions to
+generate the literalt vector to then pass to the internal propt instance.
+
+`const bvt & boolbvt::convert_bv(const exprt &expr, optionalt<std::size_t> expected_width)`
+
+Similar to convert_bitvector except it also provides basic caching and
+freezing results for incremental generation. It calls convert_bitvector 
+internally.
+
+`literalt boolbvt::convert_rest(const exprt &expr)`
+
+(Note: I'm not sure why this is split from the normal convert_bitvector, 
+but it's probably worth mentioning)
+
+`void post_process()`
+
+Performs any post-processing, which is normally adding constraints.
+
+Some key classes:
+
+\ref bv_utilst holds a set of utility functions for bit manipulation that work
+upon literalts (or vectors of them). Holds a reference to the propt 
```

Comment:
literalts .. I guess it should be literals (occurs in other places below). Or maybe keep it, but its unfortunate that doxygen won't find it referenceable (in singular, i.e. `literalt` might be turned into a valid reference).

Reply:
Wouldn't it be worth swapping `bv_utilst` description with `propt` (the first mentions the second).

Reply:
You can make Doxygen make the link by writing `[literal](\ref literalt)s`